providers/implementations/ciphers/ciphercommon_block.c

```diff
@@ -105,33 +106,32 @@ void ossl_cipher_padblock(unsigned char *buf, size_t *buflen, size_t blocksize)
         buf[i] = pad;
 }
 
+/*
+ * ossl_cipher_unpadblock removes the padding from the final block in constant time.
+ * Note that if the padding is invalid, ERR_raise is not called in order to preserve
+ * constant-timedness.
+ */
 int ossl_cipher_unpadblock(unsigned char *buf, size_t *buflen, size_t blocksize)
 {
     size_t pad, i;
     size_t len = *buflen;
+    int gp = 1;
 
-    if(len != blocksize) {
-        ERR_raise(ERR_LIB_PROV, ERR_R_INTERNAL_ERROR);
-        return 0;
-    }
+    gp &= constant_time_eq_s(len, blocksize);
```

Comment:
Initializing gp to 1 does not make sense if you want to combine it with a mask. This looks like a use of the API for which it was not designed.

Reply:
`constant_time_eq_s` returns 0xff..f if `len` == `blocksize` and 0 otherwise. So `gp` will evaluate to 1 if `len` == `blocksize` and 0 if `len` != `blocksize`. Am I missing something?

Reply:
gp will contain 0 or 1, as you intend. But the normal usage of the API is that you use a mask, and then later use a select function using the mask (to select between 0 and 1).

You're making it visible to the compiler that it can only contain 2 values. In this function it might not cause problems

Reply:
Fixed in commit `526350c`.